hack/make.ps1

```diff
@@ -319,36 +321,58 @@ Function Run-UnitTests() {
-    $goTestCommand = "go test" + $raceParm + " -cover -ldflags -w -tags """ + "autogen daemon" + """ -a """ + "-test.timeout=10m" + """ $pkgList"
+    $goTestCommand = "$GOTESTSUM_LOCATION\gotestsum.exe --format=standard-quiet --jsonfile=bundles\go-test-report-unit-tests.json --junitfile=bundles\junit-report-unit-tests.xml -- " + $raceParm + " -cover -ldflags -w -a """ + "-test.timeout=10m" + """ $pkgList"
+    Write-Host "INFO: Invoking unit tests run with $goTestCommand"
     Invoke-Expression $goTestCommand
     if ($LASTEXITCODE -ne 0) { Throw "Unit tests failed" }
 }
 
 # Run the integration tests
 Function Run-IntegrationTests() {
-    $env:DOCKER_INTEGRATION_DAEMON_DEST = $root + "\bundles\tmp"
+    $escRoot = [Regex]::Escape($root)
+    $env:DOCKER_INTEGRATION_DAEMON_DEST = $bundlesDir + "\tmp"
     $dirs = go list -test -f '{{- if ne .ForTest `"`" -}}{{- .Dir -}}{{- end -}}' .\integration\...
-    $integration_api_dirs = @()
+    #$integration_api_dirs = @()
     ForEach($dir in $dirs) {
-        $integration_api_dirs += $dir
-        Write-Host "Building test suite binary $dir"
-        go test -c -o "$dir\test.exe" $dir
-    }
-
-    ForEach($dir in $integration_api_dirs) {
+        #$integration_api_dirs += $dir
+        #Write-Host "Building test suite binary $dir"
+        #go test -c -o "$dir\test.exe" $dir
+
+        # Normalize directory name for using in the test results files.
+        $normDir = $dir.Trim()
+        $normDir = $normDir -replace $escRoot, ""
+        $normDir = $normDir -replace "\\", "-"
+        $normDir = $normDir -replace "\/", "-"
+        $normDir = $normDir -replace "\.", "-"
+        if ($normDir.StartsWith("-"))
+        {
+            $normDir = $normDir.TrimStart("-")
+            Write-Host $normDir
+        }
+        if ($normDir.EndsWith("-"))
+        {
+            $normDir = $normDir.TrimEnd("-")
+            Write-Host $normDir
+        }
+        $jsonFilePath = $bundlesDir + "\go-test-report-int-tests-$normDir" + ".json"
+        $xmlFilePath = $bundlesDir + "\junit-report-int-tests-$normDir" + ".xml"
         Set-Location $dir
         Write-Host "Running $($PWD.Path)"
         $pinfo = New-Object System.Diagnostics.ProcessStartInfo
-        $pinfo.FileName = "$($PWD.Path)\test.exe"
+        #gotestsum.exe is on the path
+        $pinfo.FileName = "gotestsum.exe"
         $pinfo.WorkingDirectory = "$($PWD.Path)"
         $pinfo.RedirectStandardError = $true
         $pinfo.UseShellExecute = $false
-        $pinfo.Arguments = $env:INTEGRATION_TESTFLAGS
+        $tmpArgs = "--format=standard-quiet --jsonfile=$jsonFilePath --junitfile=$xmlFilePath -- $env:INTEGRATION_TESTFLAGS"
+        #Write-Host $tmpArgs
+        $pinfo.Arguments = $tmpArgs
         $p = New-Object System.Diagnostics.Process
         $p.StartInfo = $pinfo
         $p.Start() | Out-Null
         $p.WaitForExit()
         $err = $p.StandardError.ReadToEnd()
+        Write-Host "$err"
```

Comment:
I _think_ this is to print `stderr` output if there's no error that happened, correct?

In case of an error (`$LASTEXITCODE -ne 0`), this will now be printed _twice_. Think it should be changed to either;

```powershell
if (($LASTEXITCODE -ne 0) -and ($err -notlike "*warning: no tests to run*")) {
    Throw "Integration tests failed: $err"
} else {
    Write-Host "$err"
}
```

Or (if `Throw` exits the script (not sure if if does so?)); perhaps add a comment as well to explain why we're printing `$err`

```powershell
if (($LASTEXITCODE -ne 0) -and ($err -notlike "*warning: no tests to run*")) {
    Throw "Integration tests failed: $err"
}

# Write stderr output
Write-Host "$err"
```